keras_cv/layers/preprocessing/random_translation.py

```diff
@@ -0,0 +1,256 @@
+            preprocessing.get_translation_matrix(tf.expand_dims(element, axis=0)),
+            fill_mode=self.fill_mode,
+            fill_value=self.fill_value,
+            interpolation=self.interpolation,
+        )
+        output = tf.squeeze(output, 0)
+        output.set_shape(image_shape)
+        return output
+
+    @staticmethod
+    def _translate_bounding_boxes_horizontal(bounding_boxes, dx):
+        x1, x2, x3, x4, rest = tf.split(
+            bounding_boxes, [1, 1, 1, 1, bounding_boxes.shape[-1] - 4], axis=-1
+        )
+        output = tf.stack(
+            [
+                x1 + dx,
+                x2,
+                x3 + dx,
+                x4,
+                rest,
+            ],
+            axis=-1,
+        )
+        output = tf.squeeze(output, axis=1)
+        return output
+
+    @staticmethod
+    def _translate_bounding_boxes_vertical(bounding_boxes, dy):
+        x1, x2, x3, x4, rest = tf.split(
+            bounding_boxes, [1, 1, 1, 1, bounding_boxes.shape[-1] - 4], axis=-1
+        )
+        output = tf.stack(
+            [
+                x1,
+                x2 + dy,
+                x3,
+                x4 + dy,
+                rest,
+            ],
+            axis=-1,
+        )
+        output = tf.squeeze(output, axis=1)
+        return output
+
+    def augment_bounding_boxes(
+        self, bounding_boxes, transformation=None, image=None, **kwargs
+    ):
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "`RandomTranslation()` was called with bounding boxes,"
+                "but no `bounding_box_format` was specified in the constructor."
+                "Please specify a bounding box format in the constructor. i.e."
+                "`RandomFlip(bounding_box_format='xyxy')`"
+            )
+
+        bounding_boxes = bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="rel_xyxy",
+            images=image,
+        )
+        bounding_boxes = RandomTranslation._translate_bounding_boxes_horizontal(
+            bounding_boxes,
+            transformation["translate_horizontal"],
+        )
+        bounding_boxes = RandomTranslation._translate_bounding_boxes_vertical(
+            bounding_boxes,
+            transformation["translate_vertical"],
+        )
+
+        bounding_boxes = bounding_box.clip_to_image(
+            bounding_boxes,
+            bounding_box_format="rel_xyxy",
+            images=image,
+        )
+        bounding_boxes = bounding_box.convert_format(
+            bounding_boxes,
+            source="rel_xyxy",
+            target=self.bounding_box_format,
+            dtype=self.compute_dtype,
+            images=image,
+        )
+        return bounding_boxes
+
+    def augment_segmentation_mask(
+        self, segmentation_mask, transformation=None, **kwargs
+    ):
+        return self._translate_image(segmentation_mask, transformation)
```

Comment:
When transforming a segmentation mask, we'll need to use `interpolation="nearest"` because segmentation masks cannot be interpolated numerically like pixels can (this could lead to erroneous values in the mask, e.g. if interpolating between class 0 and 6, you might compute class 3 which would be incorrect)

Reply:
(See a similar approach in [RandomResizedCrop](https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/preprocessing/random_resized_crop.py))